### DISCUSSION TITLE
Message Code CA1816

### DISCUSSION THREAD

**Discussion by kdub-nz** [wtertinek/Linq2Acad#42] [General]
Hi Wolfgang,
This will probably be ignorable :
From the blog site Part_3 Listing 7 public class GeneralHelper : IDisposable { . . . }
the Dispose Method is attracting an advice Message Code CA1816

>>>
Message	CA1816	Change GeneralHelper.Dispose() to call GC.SuppressFinalize(object). This will prevent derived types that introduce a finalizer from needing to re-implement 'IDisposable' to call it.	Part_3	C:\Users\KDUB\Desktop\DT_DevProjects\DT_CAD_NET8\LinqToAutoCAD_ByHand\FromTheBlog\Part_3\Commands.cs
<<<

I'm working through the Blog posts using .NET8.0 (ac2025 beta under NDA)
I don't know is this exact design is incorporated into  Linq2Acad and exactly what the ramifications, if any, will be.
The code as published performs using NET8 C#12 without visible issues.

Just thought I should comment on this.

Regards,
Kerry Brown

**Comment by wtertinek** [comment]
Hi Kerry,

thank you for bringing this up. `IDispose` is not fully implemented in the blog post code. Concerning the warning you get, calling `GC.SuppressFinalize(object)` should not be necessary, because there are no derived types of `AcadDatabase` (which is the successor of the `GeneralHelper` class in the blog post).

I have not yet had the opportunity to test Autocad 2025 and the new API. Has the API got new functions? How does .NET 8 behave? C# 12 offers many great possibilities to simplify the code of Linq2Acad.

Regards,
Wolfgang

**Reply by kdub-nz** [reply]
Hi Wolfgang,
I am wary of comment because of the NonDisclosureAgreement but I'll say I haven't had any problems.
There are a few procedural changes required in setting up projects but it becomes muscle memory quite easily.
The ObjectARX.NET libraries have no mentionable changes that I've noticed.

.NET8 and C#12 behave well ; fast and user friendly. I'm using the vs2022 Preview build which I find excellent. Hot reloads are delightful :)   I'm imaging you'd enjoy using C#12.  
transitioning from C#7.3 to C#12 isn't as difficult as I'd imagined, and there's the advantage that we don't HAVE to actually change much code if we choose, but we DO have to change to .NET8 Builds for our Assemblies and references.

Regarding the original question, your answer is as I anticipated, Thanks.

Regards,
Stay Well.
Kerry

**Reply by wtertinek** [reply]
I totally understand, thanks for the overview. I'm looking forward to using C# 12 with AutoCAD. I've already used C# 12 in other projects. It takes some time to get used to the new features.
